packages/workspace/src/browser/workspace-service.ts

```diff
@@ -353,19 +426,40 @@ export class WorkspaceService implements FrontendApplicationContribution {
      */
     async containsSome(paths: string[]): Promise<boolean> {
         await this.roots;
-        if (this._workspaceFolder) {
-            const uri = new URI(this._workspaceFolder.uri);
-            for (const path of paths) {
-                const fileUri = uri.resolve(path).toString();
-                const exists = await this.fileSystem.exists(fileUri);
-                if (exists) {
-                    return exists;
+        if (this.opened) {
+            for (const root of this._roots) {
+                const uri = new URI(root.uri);
+                for (const path of paths) {
+                    const fileUri = uri.resolve(path).toString();
+                    const exists = await this.fileSystem.exists(fileUri);
+                    if (exists) {
+                        return exists;
+                    }
                 }
             }
         }
         return false;
     }
 
+    async saveWorkspaceMetaData(uri: URI) {
```

Comment:
i would just call it `save` and have a corresponding property `get saved` to test wheter it is saved